public/components/management/groups/multiple-agent-selector.tsx

```diff
@@ -9,707 +9,897 @@
+
+        if (itemsToSave.deletedIds.length) {
+          const deleteResponse = await WzRequest.apiReq('DELETE', `/agents/group`, {
+            params: {
+              group_id: this.props.currentGroup.name,
+              agents_list: itemsToSave.deletedIds.toString(),
+            },
+          });
+          if (deleteResponse.data.data.total_failed_items) {
+            failedIds.push(...deleteResponse.data.data.failed_items);
           }
         }
-        );
-        if (deleteResponse.data.data.total_failed_items) {
-          failedIds.push(...deleteResponse.data.data.failed_items);
+
+        if (failedIds.length) {
+          const failedErrors = failedIds.map((item) => ({
+            id: ((item || {}).error || {}).code,
+            message: ((item || {}).error || {}).message,
+          }));
+
+          this.failedErrors = this.groupBy(failedErrors, 'message') || false;
+          ErrorHandler.info(
+            `Group has been updated but an error has occurred with ${failedIds.length} agents`,
+            '',
+            { warning: true }
+          );
+        } else {
+          ErrorHandler.info('Group has been updated');
         }
-      }
 
-      if (failedIds.length) {
-        const failedErrors = failedIds.map(item => ({
-          id: ((item || {}).error || {}).code,
-          message: ((item || {}).error || {}).message,
-        }));
-
-        this.failedErrors = this.groupBy(failedErrors, 'message') || false;
-        ErrorHandler.info(
-          `Group has been updated but an error has occurred with ${failedIds.length} agents`,
-          '',
-          { warning: true }
-        );
-      } else {
-        ErrorHandler.info('Group has been updated');
+        this.props.cancelButton();
+      } catch (error) {
+        this.setState({ savingChanges: false, initState: true });
+
+        //get all agents
+        let allAgents = [...this.state.availableAgents.data, ...this.state.selectedAgents.data];
+
+        //move agents to their previous position
+        allAgents.forEach((agent) => {
+          if (itemsToSave.addedIds.includes(agent.key)) {
+            this.moveItem(
+              JSON.stringify(agent),
+              this.state.selectedAgents.data,
+              this.state.availableAgents.data,
+              'a'
+            );
+          } else if (itemsToSave.deletedIds.includes(agent.key)) {
+            this.moveItem(
+              JSON.stringify(agent),
+              this.state.availableAgents.data,
+              this.state.selectedAgents.data,
+              'r'
+            );
+          }
+        });
+
+        this.checkLimit();
+
+        //relaod agent columns
+        this.reload('right');
+        this.reload('left');
+
+        try {
+          //delete nodes selected
+          await WzRequest.apiReq('DELETE', `/agents/group`, {
+            params: {
+              group_id: this.props.currentGroup.name,
+              agents_list: itemsToSave.addedIds.toString(),
```

Comment:
Deleting all agents from the `itemsToSave` list upon error means that the agents that caused the error in the first place are not in the group. So attempting to delete them causes another error. The final consequence is simply a console log, but the `failedIds` list should be updated when there is an error and used to filter what items are to be deleted.